src/common/eth-providers/consensus-provider/slots-cache.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Inject, Injectable, LoggerService } from '@nestjs/common';
+import { BlockHeaderResponse, BlockInfoResponse } from './intefaces';
+import { LOGGER_PROVIDER } from '@lido-nestjs/logger';
+
+export interface CachedSlot {
+  missed: boolean;
+  header?: BlockHeaderResponse | void;
+  info?: BlockInfoResponse | void;
+}
+
+@Injectable()
+export class SlotsCacheService {
+  constructor(@Inject(LOGGER_PROVIDER) protected readonly logger: LoggerService) {}
+
+  private cache: { [slot: string]: CachedSlot } = {};
+
+  public put(slot: string, cache: CachedSlot): void {
+    this.cache[slot] = cache;
+  }
+
+  public update(slot: string, fresh: CachedSlot): void {
+    const old = this.get(slot) ?? {};
+    this.cache[slot] = { ...old, ...fresh };
+  }
+
+  public get(slot: string): CachedSlot {
+    return this.cache[slot];
+  }
+
+  /**
+   * Purge old slots from cache
+   * @param epoch - current processed epoch
+   */
+  public purgeOld(epoch: bigint): void {
+    let purged = 0;
+    const firstSlotPrev = (epoch - 1n) * 32n;
+    for (const slot of Object.keys(this.cache)) {
+      // remove the oldest slots and all hashes
```

Comment:
There is only one entity type - Slot (it can be missed or not. If not missed - has header and info). But sometimes we need to get block info by its root. Managing the lifetime of such records is too expensive, and we find it easier to delete these records, since there is a high probability that we won't need them in the next cycle

Reply:
Just give a descriptive type for that and it will be fine.

Reply:
done